src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -32,14 +32,14 @@ public class HubConnection
         private HubProtocolReaderWriter _protocolReaderWriter;
 
         private readonly object _pendingCallsLock = new object();
-        private readonly CancellationTokenSource _connectionActive = new CancellationTokenSource();
         private readonly Dictionary<string, InvocationRequest> _pendingCalls = new Dictionary<string, InvocationRequest>();
         private readonly ConcurrentDictionary<string, List<InvocationHandler>> _handlers = new ConcurrentDictionary<string, List<InvocationHandler>>();
+        private CancellationTokenSource _connectionActive;
 
         private int _nextId = 0;
         private volatile bool _startCalled;
 
-        public Task Closed { get; }
+        public event Action<Exception> Closed;
```

Comment:
Shouldn't this be a wrapper around `_connection.Closed`?

Reply:
No. This way the only event that is hooked to HttpConnection is the one from hubConnection and the user subscribes to the object they are using.